cmd/jaeger/internal/integration/es_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	elasticsearch8 "github.com/elastic/go-elasticsearch/v8"
+	"github.com/olivere/elastic"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+type ESStorageIntegration struct {
+	E2EStorageIntegration
+
+	client   *elastic.Client
+	v8Client *elasticsearch8.Client
+	logger   *zap.Logger
+}
+
+const (
+	host                     = "0.0.0.0"
+	queryPort                = "9200"
+	queryHostPort            = host + ":" + queryPort
+	queryURL                 = "http://" + queryHostPort
+	indexPrefix              = "integration-test"
+	indexDateLayout          = "2006-01-02"
+	tagKeyDeDotChar          = "@"
+	maxSpanAge               = time.Hour * 72
+	defaultMaxDocCount       = 10_000
+	spanTemplateName         = "jaeger-span"
+	serviceTemplateName      = "jaeger-service"
+	dependenciesTemplateName = "jaeger-dependencies"
+	primaryNamespace         = "es"
+	archiveNamespace         = "es-archive"
+)
+
+func (s *ESStorageIntegration) initializeES(t *testing.T) {
```

Comment:
All of this functionality already exists in the other es test. Try to find a way to reuse it.